src/main/java/org/libertybikes/ai/AILogic.java

```diff
@@ -41,118 +41,58 @@
         OPEN,
         FIXED_OBSTACLE,
         MOVING_OBSTACLE,
-        PLAYER
+        PLAYER,
+        PLAYER_TRAIL
     }
 
     private final GAME_SLOT[][] gameBoard = new GAME_SLOT[BOARD_SIZE][BOARD_SIZE];
-    private final String myPlayerId;
-    private boolean filledFixedObstacles = false;
-    private DIRECTION currentDirection = DIRECTION.RIGHT;
-    private int currentX;
-    private int currentY;
 
     public AILogic(String myId) {
-        myPlayerId = myId;
         for (int x = 0; x < BOARD_SIZE; x++)
             Arrays.fill(gameBoard[x], GAME_SLOT.OPEN);
     }
 
     /*
-     * This should contain the main logic of the AI, it receives a JSON straing of all the
+     * This should contain the main logic of the AI, it receives a JSON string of all the
      * objects on the board, players and obstacles.
      */
     public DIRECTION processAiMove(GameTick gameTick) {
-        updateBoard(gameTick);
-
-        System.out.println("Currently facing: " + currentDirection);
-        DIRECTION bestDirection = currentDirection;
-        int bestDirectionSlots = 0;
-        for (DIRECTION d : DIRECTION.values()) {
-            int freeSlots = freeSlots(d);
-            if (freeSlots > bestDirectionSlots) {
-                bestDirection = d;
-                bestDirectionSlots = freeSlots;
-            }
-        }
-        currentDirection = bestDirection;
-        System.out.println("  will face: " + currentDirection);
-        return currentDirection;
-    }
-
-    private int freeSlots(DIRECTION lookingDirection) {
```

Comment:
lets leave this method in place (along with any require class-level fields such as `currentX`)